File: go/2391.go
func garbageCollection(garbage []string, travel []int) int {
    minutes := 0

    var lastG, lastP, lastM, length int = -1,-1,-1,0

    for i,s := range garbage {
        length += len(s)
        if strings.Contains(s, "G") {
            lastG = i
        }
        if strings.Contains(s, "P") {
            lastP = i
        }
        if strings.Contains(s, "M") {
            lastM = i
        }
    }

    if lastG != -1 {
        for i := 0; i < lastG; i++ {
            minutes += travel[i]
        }
    }
    if lastM != -1 {
        for i := 0; i < lastM; i++ {
            minutes += travel[i]
        }
    }
    if lastP != -1 {
        for i := 0; i < lastP; i++ {
            minutes += travel[i]
        }
    }

    return minutes + length
}

File: py/2391.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        minutes = 0

        lastG, lastP, lastM = -1,-1,-1
        length = 0
        for i,s in enumerate(garbage):
            length += len(s)
            if 'G' in s:
                lastG = i
            if 'P' in s:
                lastP = i
            if 'M' in s:
                lastM = i

        if lastG != -1:
            minutes += sum(travel[:lastG])
        if lastP != -1:
            minutes += sum(travel[:lastP])
        if lastM != -1:
            minutes += sum(travel[:lastM])

        return minutes + length
